Content.Server/Traits/Assorted/NarcolepsyComponent.cs

```diff
@@ -20,5 +20,21 @@ public sealed partial class NarcolepsyComponent : Component
     [DataField("durationOfIncident", required: true)]
     public Vector2 DurationOfIncident { get; private set; }
 
+    [DataField] // # DeltaV begin Narcolepsy port from EE
     public float NextIncidentTime;
+
+    /// <summary>
+    ///     Locales for popups shown when the entity is about to fall asleep/is waking up.
+    ///     They are fetched in the format of "(base)-(random number between 1 and count)", e.g. "narcolepsy-warning-popup-3".
+    /// </summary>
+    [DataField]
+    public string WarningLocaleBase = "narcolepsy-warning-popup", WakeupLocaleBase = "narcolepsy-wakeup-popup";
```

Comment:
this should just use a localized dataset proto instead

Reply:
could you spare an example of how this is done in another component for this poor shitcoder

Reply:
Just specify it in the comp
https://github.com/DeltaV-Station/Delta-v/blob/177f4e1568edd3ad46aa23e4d7ccda7d5eb6ab05/Content.Server/DeltaV/StationEvents/Components/FugitiveRuleComponent.cs#L56-L60

 and then pick a random one from the dataset in a separate helper method

```cs
    private string GetRandomPopup()
    {
        return Loc.GetString(_random.Pick(_prototypeManager.Index(comp.Dataset).Values));
    }
```